### ISSUE TITLE
Can't open object-folder in manyToOne-relation if folder is assigned with script

### ISSUE THREAD

**Issue by MosiiAUT** [pimcore/pimcore#7571]
### Expected behavior
Assign folder via php to a objects manyToOne-relation field and then open it by clicking on the relation or the "open folder"-icon.
### Actual behavior
Assigning works without problems, relations are ok. If you try to open it, it is trying to open the folder via /get instead of /get-folder and will throw this error:

**Argument 1 passed to Pimcore\Bundle\AdminBundle\Controller\Admin\DataObject\DataObjectController::getLatestVersion() must be an instance of Pimcore\Model\DataObject\Concrete, instance of Pimcore\Model\DataObject\Folder given, called in /home/protina/www/vendor/pimcore/pimcore/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php on line 366**

If you then assign the folder again per hand in the Pimcore-Backend, opening it is no problem anymore.
Asset-Folders are also no problem. I don't about Document-Folders :)

### Steps to reproduce
Create an object with a manyToOne-relation (doesn't matter if restricted only to folders) and then assign a object-folder via a php-script.

### Code example
```php
$testFolder = Object\Folder::getById(21);
$testObj->setTestFolder($testFolder);
$testObj->save();
```

**Comment by weisswurstkanone** [comment]
Unable to reproduce on latest version. @markus-moser which release you are on?

**Comment by MosiiAUT** [comment]
Tested on 6.8.4 and 6.5.3

**Comment by dvesh3** [comment]
I can reproduce it by dbl clicking on the field..

**Comment by weisswurstkanone** [comment]
for reference: https://github.com/pimcore/pimcore/blob/15af69103317a0a264ea48227432eadc9e33e892/models/DataObject/Concrete/Dao.php#L110

**Comment by dvesh3** [comment]
Fixed by #7578
